GeneratorInterface/GenFilters/plugins/LHEPtFilter.cc

```diff
@@ -0,0 +1,121 @@
+// -*- C++ -*-
+//
+// Package:    LHEPtFilter
+// Class:      LHEPtFilter
+//
+/* 
+
+ Description: Filter to select events with pT in a given range.
+ (Based on LHEGenericFilter)
+
+     
+*/
+//
+
+// system include files
+#include <memory>
+#include <iostream>
+#include <set>
+
+// user include files
+#include "Math/Vector4D.h"
+#include "Math/Vector4Dfwd.h"
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDFilter.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "SimDataFormats/GeneratorProducts/interface/LHEEventProduct.h"
+
+//
+// class declaration
+//
+
+class LHEPtFilter : public edm::EDFilter {
+public:
+  explicit LHEPtFilter(const edm::ParameterSet&);
+  ~LHEPtFilter() override;
+
+private:
+  bool filter(edm::Event&, const edm::EventSetup&) override;
+  void endJob() override;
+
+  // ----------member data ---------------------------
+
+  edm::EDGetTokenT<LHEEventProduct> src_;
+  std::vector<lhef::HEPEUP::FiveVector> lheParticles;
+  std::vector<ROOT::Math::PxPyPzEVector> cands;
```

Comment:
By quick look it looks like `lheParticles` and `cands` are not needed as member variables.

Reply:
Right, have switched these to local variables now.